tests/TreeHouse/EventStore/Tests/DBALEventStoreTest.php

```diff
@@ -320,21 +314,50 @@ public function it_upcasts_events()
         );
 
         $someUpcaster = new DummyUpcaster(function (SerializedEvent $e) {
-            return $e->getName() == 'Anything' && $e->getPayloadVersion() == 1;
+            return $e->getName() === 'Anything' && $e->getPayloadVersion() === 1;
         }, $upcastedEvent);
 
+        $multipleUpcastedEvents = [
+            $upcastedEvent = new SerializedEvent(
+                $upcastedEvent->getId(),
+                $upcastedEvent->getName(),
+                '{"payload":"json", "with": "extra"}',
+                $upcastedEvent->getPayloadVersion() + 1,
+                1,
+                new DateTime('2015-01-01 00:00:00')
+            ),
+            $extraUpcastedEvent = new SerializedEvent(
+                $upcastedEvent->getId(),
+                'ExtraAnything',
+                '{"data": true}',
+                1,
+                1,
+                new DateTime('2015-01-01 00:00:00')
+            ),
+        ];
+
+        $someMultiUpcaster = new DummyUpcaster(function (SerializedEvent $e) {
+            return $e->getName() === 'Anything' && $e->getPayloadVersion() === 2;
+        }, $multipleUpcastedEvents);
+
         $chain = new SimpleUpcasterChain();
         $chain->registerUpcaster($someUpcaster);
+        $chain->registerUpcaster($someMultiUpcaster);
 
         // test
 
         $eventStore->setUpcaster($chain);
 
-        $stream = $eventStore->getStream(self::UUID);
+        $stream = iterator_to_array($eventStore->getStream(self::UUID));
 
         $this->assertEquals(
             $eventFactory->createFromSerializedEvent($upcastedEvent),
-            iterator_to_array($stream)[0]
+            $stream[0]
+        );
+
+        $this->assertEquals(
+            $eventFactory->createFromSerializedEvent($extraUpcastedEvent),
+            $stream[1]
```

Comment:
Instead of modifying this test, you could add a second test `it_upcasts_events_into_multiple_events`. That way it is clear that the old way is still supported.